api/client-server/cross_signing.yaml

```
@@ -0,0 +1,215 @@
+# Copyright 2020 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+swagger: '2.0'
+info:
+  title: "Matrix Client-Server Cross Signing API"
+  version: "1.0.0"
+host: localhost:8008
+schemes:
+  - https
+  - http
+basePath: /_matrix/client/%CLIENT_MAJOR_VERSION%
+consumes:
+  - application/json
+produces:
+  - application/json
+securityDefinitions:
+  $ref: definitions/security.yaml
+paths:
+  "/keys/device_signing/upload":
+    post:
+      summary: Upload cross-signing keys.
+      description: |-
+        Publishes cross-signing keys for the user.
+
+        This API endpoint uses the `User-Interactive Authentication API`_.
+      operationId: uploadCrossSigningKeys
+      security:
+        - accessToken: []
+      parameters:
+        - in: body
+          name: keys
+          description: |-
+            The keys to be published.
+          schema:
+            type: object
+            properties:
+              master_key:
+                description: |-
+                  Optional. The user\'s master key.
+                allOf:
+                  - $ref: definitions/cross_signing_key.yaml
+              self_signing_key:
+                description: |-
+                  Optional. The user\'s self-signing key. Must be signed with
```

consistent use of with/by would be nice.
```suggestion
                  Optional. The user\'s self-signing key. Must be signed by
```